src/rt.rs

```diff
@@ -13,84 +13,90 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with rgtk.  If not, see <http://www.gnu.org/licenses/>.
 
-use libc::c_uint;
 use std::ptr;
 use ffi;
 use glib::translate::{FromGlibPtr};
 use glib::{to_bool, to_gboolean};
 
-pub fn init() {
-    unsafe {
-        ffi::gtk_init(ptr::null(), ptr::null());
+pub struct Gtk;
+
+impl Gtk {
+    pub fn new() -> Gtk {
```

Comment:
Umm, I think if we go down the runtime checks path, we'll just panic when you try to touch GTK from a wrong thread. There could be a function to find out if the thread is the right one.